### ISSUE TITLE
Warn on missing binding flags or method/property signature when the type or member name is unknowable

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/ReflectionAnalyzers#63]
Also, no binding flags warning is given here where the type is unknowable to the analyzer, but it's still important to make the user pick them.

For the binding flags, the fix would be to add `Public | Static | Instance` to keep the same behavior but make it explicit.

The signature is unknowable, so that would trigger a warning with no fix. (Don't forget properties and constructors!)

**Comment by JohanLarsson** [comment]
Separate analyzer so it can be disabled if annoying? I'm leaning towards default disabled or hidden.

**Comment by jnm2** [comment]
I don't see any chance that I'd want to suppress the warning to be explicit any more than if the type was known by the analyzer.

**Comment by JohanLarsson** [comment]
Having it separate allows for configuring it but that may also be annoying, dunno.

**Comment by jnm2** [comment]
True. I'm not sure why a knowable vs unknowable type makes a difference in what's being reported. Either way, you can see for sure whether or not the person is using binding flags.

**Comment by brian-reichle** [comment]
I'm a bit hesitant about this, there have been cases where the name, declaring type & signature were not known at compile time, but rather the name/declaring type were provided at runtime and the arguments that needed to be passed in were calculated from the method signature.

**Comment by jnm2** [comment]
Good note. What about at least binding flags and then a separate, confirgurable warning for types?

**Comment by brian-reichle** [comment]
I'm fine with always providing binding flags explicitly.

**Comment by jnm2** [comment]
I do want to be able to opt in to a warning when explicit types are not provided for a method or property that the analyzer will not be able to find in metadata.
